libraries/go/webhook.go

```diff
@@ -42,7 +56,46 @@ func NewWebhook(secret string) (*Webhook, error) {
 		return nil, fmt.Errorf("unable to create webhook, err: %w", err)
 	}
 	return &Webhook{
-		key: key,
+		key:           key,
+		signingMethod: "hmac",
+	}, nil
+}
+
+func NewAsymmetricVerificationWebhook(publicKey string, options *WebhookOptions) (*Webhook, error) {
```

Comment:
I would generify this a bit. Have something like:

```go
func NewWebhookWithOptions(method WebhookMethod, options *WebhookOptions)
```

And `WebhookMethod` only allows a specific enum value. Then, `options` can take in a crypto key object for the sign / verify use case, so decoding below does not need to happen.

Reply:
Also, move the `NewWebhook` implementation to use this generic one with HMAC.

Reply:
Gotcha. `NewWebhook` will call `NewWebhookWithOptions(HMAC, options)` under the hood. 

Quick confirmation: would this crypto key object be any / untyped? Figure that the signing object for each  encryption method (e.g. ed25519 / RSA) would have slightly different params.

Reply:
Ah I guess this would be any object that satisfies the CryptoKey interface we define